c2/cuda/groth16_cuda.cu

```diff
@@ -431,11 +490,24 @@ RustError::by_value generate_groth16_proofs_c(const Assignment<fr_t> provers[],
             if (b_bit_off)
                 b_bit_vector[b_bits_cursor] = b_bits;
 
-            assert(points_a_cursor == points_a.size());
-            assert(points_b_cursor == points_b_g1.size());
-            assert(a_cursor == a_popcount);
-            assert(b_cursor == b_popcount);
-            assert(l_cursor == l_popcount);
+            if (!l_skip_batch_add)
+                assert(l_cursor == l_popcount);
+            else
+                assert(l_cursor == points_l.size());
+
+            if (!a_skip_batch_add) {
+                assert(points_a_cursor == points_a.size());
+                assert(a_cursor == a_popcount);
+            }
+            else
+                assert(a_cursor == points_a.size());
```

Comment:
Customarily it's said that if `if` has curly braces, then `else` does too, even if not required. [And vice versa actually.] And what's up with else on the new line? Who does it:-)

Reply:
> And what's up with else on the new line? Who does it:-)

I had always done it. It's easier for me to read when `}` and `else` are on new lines.

Reply:
> I had always done it.

It doesn't mean that it's right:-):-):-) But on a serious note...

> It's easier for me to read when `}` and `else` are on new lines.

Readability is actually more about others than yourself. It's about how you **communicate** the idea.

Reply:
I'm kind of struggling with !x_skip_batch_add all over. I mean it feels like a negative would be a better choice... You would naturally have to change the names. How about x_split_msm? So that conditions would be `if (x_split_msm)`...

Reply:
Sure

Reply:
> Readability is actually more about others than yourself.

Fair enough